### PR TITLE
Version Upgrades & FE Build Process Improvements

### PR DESCRIPTION
**Pull request by mtrythall**
Changes proposed in this PR:

- Use Parcel JS as the FE compiler instead of Gulp
- Updates documentation to reflect Parcel usage
- Removes versioned Foundation, Headroom, jQuery, etc from static/ folder
- Updates HTML templates to reference new files
- Bumps all versions of FE dependencies to latest versions
- Updates dot files to reflect changes
- Removes references to surge.sh as the project is using Netlify
- Removes style sherpa, the styleguide builder, as we'll be moving off of it in future work
- Bump Ruby version to 2.6.1: https://www.ruby-lang.org/en/news/2019/01/30/ruby-2-6-1-released/

This doesn't touch the front-end of the site, and things there will be wonky with the exception of the landing page (should still look the same). It's an under-the-hood change.

### CODE REVIEW

**Comment by mtrythall** [comment]
Note, the Ruby version bump is optional. Just felt like a good thing to do.

**Comment by mtrythall** [comment]
Branch deploys but CSS and JS aren't compiled. We'll need to run `npm run build` which equates to `parcel build main.js -d static/ --no-source-maps`

**Comment by mtrythall** [comment]
Thanks. Didn't catch that.

**Comment by mtrythall** [comment]
Though we're not getting styles during deployment:

https://deploy-preview-16--djangoconus2019.netlify.com/conduct/

@jefftriplett Where do we adjust the commands needed to compile CSS/JS?

File: _layouts/base.html [hunk 1]
```diff
@@ -11,7 +11,7 @@
 {% include youtube-embed-head.html id=youtube_id %}
 {% endif %}
     <link href="https://fonts.googleapis.com/css?family=Source+Sans+Pro:400,400i,700,700i|Signika:600,700" rel="stylesheet">
-    <link rel="stylesheet" href="/static/css/styles.css">
+    <link rel="stylesheet" href="/static/main.css">
```
Comment (jefftriplett) [inline _layouts/base.html]: @mtrythall this path change is what broke the css outside of the homepage. The homepage only uses inline styles from what I'm seeing. 

I'm updating my local setup to test and then I'll change it back unless you updated this for some other reason.

Comment (jefftriplett) [reply _layouts/base.html]: @mtrythall on second thought, I think we are seeing caching.

**Pull request merged**